### PR TITLE
REF: implement _reconstruct_ea_result, _ea_to_cython_values

### PR DESCRIPTION
**Pull request by jbrockmendel**
The idea being to eventually refactor those into EA/EADtype methods, xref #43682

### CODE REVIEW

File: pandas/_libs/groupby.pyx [hunk 1]
```diff
@@ -1350,6 +1350,10 @@ cdef group_min_max(iu_64_floating_t[:, ::1] out,
                     else:
                         if uses_mask:
                             result_mask[i, j] = True
+                            # set out[i, j] to 0 to be deterministic, as
```
Comment (jreback) [inline pandas/_libs/groupby.pyx:1353]: shouldn't you just np.zeros instead of empyt to start?

Comment (jbrockmendel) [reply pandas/_libs/groupby.pyx:1353]: empty is cheaper in the non-masked case

**Comment by jreback** [comment]
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=74625&view=results looks like 32 bit failing

**Comment by jbrockmendel** [comment]
> looks like 32 bit failing

fixed

**Pull request merged**